lib/ops/opsservice/service.go

```diff
@@ -379,12 +381,13 @@ func (o *Operator) GetExpandToken(key ops.SiteKey) (*storage.ProvisioningToken,
 	}
 
 	for _, token := range tokens {
-		if token.Type == storage.ProvisioningTokenTypeExpand {
+		// return long-lived join token
+		if token.Type == storage.ProvisioningTokenTypeExpand && token.Expires.IsZero() {
```

Comment:
We would need to implement this feature specifically. This function basically returns token displayed by "gravity status" so I've just updated it to be 100% sure it will return long-lived token.